implementations/elixir/ockam/ockam_abac/lib/attribute_rules.ex

```diff
@@ -0,0 +1,239 @@
+defmodule Ockam.ABAC.AttributeRule do
+  @moduledoc """
+  Attribute rule matching AST for Ockam.ABAC
+  """
+
+  alias Ockam.ABAC.AttributeRule.Formatter
+
+  alias Ockam.ABAC.Request
+
+  ## Only binaries for now
+  ## TODO: support more types for lt/gt
+  @type attribute_source() :: :resource | :action | :subject
+  @type key() :: {source :: attribute_source(), name :: binary()}
+  @type value() :: binary()
+
```

Comment:
wonder if a an expression language with terminals, literals, etc won't be easier to reason about long term. And introduce typed values.    Made an attempt [here](https://gist.github.com/polvorin/9f49fd92c1462f433e5ed2e6d047882c)

Reply:
Also I've removed all usages of `AttributeRule.new` from the tests. Let's keep the internal implementation opaque for the user and only suggest to use the s-expression format.